bifrost.go

```diff
@@ -271,3 +271,42 @@ func newWasabiCloudStorage(bc *BridgeConfig) (RainbowBridge, error) {
 		UseAsync:       bc.UseAsync,
 	}, nil
 }
+
+// DeleteGCSobject deletes an object from one or more buckets
+func DeleteGCSobject(bc *BridgeConfig) error {
+
+	var client *storage.Client
+	var err error
+	if bc.CredentialsFile != "" {
+		// first attempt to authenticate with credentials file
+		client, err = storage.NewClient(context.Background(), option.WithCredentialsFile(bc.CredentialsFile))
+		if err != nil {
+			return &errors.BifrostError{
+				Err:       err,
+				ErrorCode: errors.ErrUnauthorized,
+			}
+		}
+	} else {
+		// if no credentials file is specified, attempt to authenticate without credentials file
+		client, err = storage.NewClient(context.Background())
+		if err != nil {
+			return &errors.BifrostError{
+				Err:       err,
+				ErrorCode: errors.ErrUnauthorized,
+			}
+		}
+	}
+
+	for _, bucketName := range bc.Buckets {
+		bucket := client.Bucket(bucketName)
+		obj := bucket.Object(bc.Object)
+
+		if err := obj.Delete(context.Background()); err != nil {
+			return &errors.BifrostError{
+				Err:       err,
+				ErrorCode: errors.ErrUnauthorized,
+			}
+		}
+	}
+	return nil
+}
```

Comment:
thank you for the contribution but this is not how bifrost works. 

bifrost implements an interface called the RainbowBridge that's shared across every provider ensuring that you call the exact same function irrespective. 

you current implementation negates that, if we want to delete an S3 object, this would mean we have to give it a different name. 


you are to check inisde the gcs folder, you'll find the file that holds the functions for the RainbowBridge interface for gcs. You need to create a new function on it called DeleteFile which should take input in the form of the UploadFile

you'll notice that the moment you add this functions, errors will appear because the struct for gcs will no longer implement the RainbowBridge interface. You'd have to fix this by updating the interface to support DeleteFile and also updating all other provider's struct to support the function...they really don't need to do anything, they can just return nil.

Reply:
Got it. 
I will adjust the implementation. Thank you for your feedback.